test/e2e/util.go

```diff
@@ -1466,15 +1467,26 @@ func waitForNodeToBe(c *client.Client, name string, wantReady bool, timeout time
 }
 
 // checks whether all registered nodes are ready
-func allNodesReady(c *client.Client) error {
-	nodes, err := c.Nodes().List(labels.Everything(), fields.Everything())
-	Expect(err).NotTo(HaveOccurred())
+func allNodesReady(c *client.Client, timeout time.Duration) error {
+	Logf("Waiting up to %v for all nodes to be ready", timeout)
+
 	var notReady []api.Node
-	for _, node := range nodes.Items {
-		if isNodeReadySetAsExpected(&node, false) {
-			notReady = append(notReady, node)
+	err := wait.Poll(poll, timeout, func() (bool, error) {
+		notReady = nil
+		nodes, err := c.Nodes().List(labels.Everything(), fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
Isn't it better to just retry?  By omitting this, and instead returning false,err I think you'll be more resilient.